modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java

```diff
@@ -1002,6 +1005,43 @@ public boolean created() {
     @Override protected CorruptedTreeException corruptedTreeException(String msg, Throwable cause, int grpId, long... pageIds) {
         CorruptedTreeException e = new CorruptedTreeException(msg, cause, grpName, cacheName, idxName, grpId, pageIds);
 
+        String errorMsg = "Index of the table " + tblName + " is corrupted, to fix this issue a rebuild " +
```

Comment:
I think it is necessary to specify not only the table name, but also the index name.